theano/gof/op.py

```diff
@@ -783,6 +783,9 @@ def __new__(cls, *args, **kwargs):
         # this function exists to silently and transparently ensure that all
         # existing Ops get a _op_use_c_code attribute
         obj = object.__new__(cls)
+        # Ensuring all the Ops created henceforth has __props__
+        if not hasattr(obj, '__props__'):
+            raise
```

Comment:
I'm not sure it is the right place for this. Did you check that __props__ is already there for some ops that already have it? I think it?

In all case, I don't want to make props mandatory for now. But if all ops in Theano have it, we could raise a warning here.